lib/ansible/playbook/role/requirement.py

```diff
@@ -190,6 +190,17 @@ def scm_archive_role(src, scm='git', name=None, version='HEAD'):
         if rc != 0:
             raise AnsibleError ("- command %s failed in directory %s (rc=%s)" % (' '.join(clone_cmd), tempdir, rc))
 
+        if scm == 'git' and version:
+            checkout_cmd = [scm, 'checkout', version]
+            with open('/dev/null', 'w') as devnull:
+                try:
+                    popen = subprocess.Popen(checkout_cmd, cwd=os.path.join(tempdir, name), stdout=devnull, stderr=devnull)
+                except (IOError, OSError):
+                    raise AnsibleError("error executing: %s" % " ".join(checkout_cmd))
+                rc = popen.wait()
+            if rc != 0:
```

Comment:
i'm fine with the less 'zenliness' of `rc != 0` as this reflects the non pythonic shelling out to a command line utility